includes/Core/User_Surveys/REST_User_Surveys_Controller.php

```diff
@@ -137,9 +149,23 @@ protected function get_rest_routes() {
 						$data         = $request->get_param( 'data' );
 
 						$response = $proxy->send_survey_event( $creds, $access_token, $data['session'], $data['event'] );
-						$response = rest_ensure_response( $response );
 
-						return $response;
+						if ( ! is_wp_error( $response ) ) {
+							$is_survey_closed    = isset( $data['event']['survey_closed'] );
+							$is_completion_shown = isset( $data['event']['completion_shown'] );
+							if ( $is_completion_shown || $is_survey_closed ) {
+								$survey = $this->queue->find_by_session( $data['session'] );
+								if ( ! empty( $survey ) ) {
+									$this->queue->dequeue( $survey['survey_id'] );
+								}
+							}
+
+							if ( isset( $data['event']['survey_shown'] ) ) {
+								$this->timeouts->add( '__global', 12 * HOUR_IN_SECONDS );
```

Comment:
How about we extract this to a new method like `set_global_timeout()`? Then we don't need to use the special global name, or the timeout value since this would always be the same (we could always pass that as a parameter though and use this as the default). Alternatively we could use a constant for the global ID which is probably a good idea too but tucking it in a method should discourage reaching for it directly.

Reply:
Yes, sounds good to me. Added the `set_global_timeout` method.